pages/1.13/security/oss/user-account-management/service-accounts/index.md

```diff
@@ -0,0 +1,76 @@
+---
+layout: layout.pug
+navigationTitle: Service Accounts
+title: Service Account Management
+excerpt: Managing service accounts
+menuWeight: 30
+---
+
+<!-- The source repository for this topic is https://github.com/dcos/dcos-docs-site -->
+
+# Add a service account
+
+## Using the IAM API
+
+**Prerequisite:**
+- [OpenSSL](https://www.openssl.org/)
+- [DC/OS Authentication token](/1.13/security/oss/authentication/authentication-token/) exported into the environment as `TOKEN`.
+
+A service account consists of a user ID and a RSA private key.
+
+1. To add a service account using the DC/OS [Identity and Access Management (IAM) API](/1.13/security/oss/iam-api/) generate a RSA private key first using OpenSSL.
+
+```bash
+openssl genpkey -algorithm RSA -out private-key.pem -pkeyopt rsa_keygen_bits:2048
+```
+
+2. Extract the corresponding public key from the private key.
+
+```bash
+openssl rsa -pubout -in private-key.pem -out public-key.pem
+```
+
+3. Then replace `<service-account-id>` with the desired value in the following command and execute it:
+
+```bash
+curl -i -X PUT http://<host-ip>/acs/api/v1/users/<service-account-id> -d '{"public_key": "'"$(sed ':a;N;$!ba;s/\n/\\n/g' public-key.pem)"'", "provider_type": "internal"}' -H 'Content-Type: application/json' -H "Authorization: token=$TOKEN"
+```
```

Comment:
Can you leave out provider_type as you did for regular user accounts?

Reply:
Can you explain the `'"$(sed ':a;N;$!ba;s/\n/\\n/g'` stunt in simple words so that it's easy to review? :)

Reply:
Changing to `sed ':a;N;s/\n/\\n/g;ta'` which is a simpler version of the former.
What it does: It creates a conditional loop over all lines of a file to replace newlines in each line.
The `:a` creates a branch label. `N` indicates operation on one line. `s/x/y/g` replaces x with y. `ta` is a conditional jump to label `a` if the regex condition was met.
```
[tim@carbon] ~/Tmp cat test
hello
world
again
[tim@carbon] ~/Tmp cat test | sed ':a;N;s/\n/\\n/g;ta'
hello\nworld\nagain
```

Reply:
Should we add something along these lines to the text? When dealing with credentials, a good administrator would be wary of a command that does something like this without explanation.

Also, it allows them to translate the `sed` command to an equivalent in another scripting language.